src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs

```diff
@@ -417,6 +418,110 @@ public async Task CreateOrUpdatePullRequestCommentAsync(string pullRequestUrl, s
+            {
+                MergePolicyEvaluationResult.SingleResult remainingCheckCommon = evaluations.SingleOrDefault(eval => remainingCheck.ExternalId == $"maestro-policy-{eval.MergePolicyName}-{latestCommit.Sha}");
+                // Avoid deleting check(s) that aren't from maestro
+                if (remainingCheckCommon != null)
+                {
+                    CheckRunUpdate updatedCheck = new CheckRunUpdate();
+                    updatedCheck.Status = "completed";
+                    updatedCheck.Conclusion = "skipped";
+                    updatedCheck.Name = remainingCheck.Name;
+                    updatedCheck.CompletedAt = remainingCheck.CompletedAt;
+                    await Client.Check.Run.Update(owner, repo, remainingCheck.Id, updatedCheck);
+                }
+            }
+        }
+
+        private NewCheckRun CreateNewCheckRun(MergePolicyEvaluationResult.SingleResult result, string sha)
+        {
+
+            // TODO: Investigate this condition, in which situation can this happen (and what we should do if that happens) ?
+            // Initially we just returned a faile status with the result message
+            if (result.MergePolicyName == null)
+            {
+                var newCheck = new NewCheckRun($"{result.MergePolicyDisplayName}", sha);
+                var output = new NewCheckRunOutput(result.MergePolicyName, result.Message);
+                newCheck.Output = output;
+                newCheck.ExternalId = $"maestro-policy-{result.MergePolicyName}-{sha}";
+                newCheck.Status = CheckStatus.Completed;
+                newCheck.Conclusion = "failure";
+                newCheck.CompletedAt = DateTime.Now;
+                return newCheck;
+            }
+
+            if (result.Success == null)
+            {
+                var newCheck = new NewCheckRun($"{result.MergePolicyDisplayName}", sha);
+                var output = new NewCheckRunOutput(result.MergePolicyName, result.Message);
+                newCheck.Output = output;
+                newCheck.ExternalId = $"maestro-policy-{result.MergePolicyName}-{sha}";
+                newCheck.Status = CheckStatus.InProgress;
+                return newCheck;
+
+            }
+
+            if (result.Success == true)
+            {
+                var newCheck = new NewCheckRun($"{result.MergePolicyDisplayName}", sha);
+                var output = new NewCheckRunOutput(result.MergePolicyName, result.Message == null ? "" : $" - {result.Message}");
+                newCheck.Output = output;
+                newCheck.ExternalId = $"maestro-policy-{result.MergePolicyName}-{sha}";
+                newCheck.Status = CheckStatus.Completed;
+                newCheck.Conclusion = "success";
+                newCheck.CompletedAt = DateTime.Now;
+                return newCheck;
+
+            }
+            var newCheckRun = new NewCheckRun($"{result.MergePolicyDisplayName} {result.Message}", sha);
```

Comment:
newCheckRun should always have name of result.MergePolicyDisplayName - without result.Message.
This will result into `var newCheck = new NewCheckRun($"{result.MergePolicyDisplayName}", sha);` be same in all cases and hence could be removed from these cases and moved above in higher scope. (example line 476)